### ISSUE TITLE
[APPENG-6023] Replace rule-based ran-anomaly-detector with Mantis binary AD (TelecomTS)

### ISSUE THREAD

**Issue by TomerG2** [rh-ai-quickstart/ai-driven-network-remediation#147]
## Problem Statement

The Telco/O-RAN workflow detects anomalies with a **rule engine** on a **six-KPI simulated cell stream** (`LowRsrp`, `ThroughputDrop`, and similar). Operators and demo audiences never see **ML-based binary anomaly detection**, even though a fine-tuned Mantis-8M AD head on **TelecomTS** exists and OpenShift AI model serving is a core platform story for this quickstart.

Rules cannot score the time-series patterns that model was trained on. The random six-KPI factory is not TelecomTS. Without an AD stage served on OpenShift AI, the agent cannot demonstrate **train → serve → consume** for detection, and sibling APPENG-6062 has no honest window to classify.

## Solution

Replace rule-based detection with **binary Mantis AD** on **TelecomTS lab-trace fixtures**. Demo data is public 5G observability samples (18 channels × 128 timesteps), said out loud as lab replay, not a live multi-cell RAN.

**On-demand injection** stays `POST /api/demo/trigger` on `ran-chatbot-service` (frontend buttons). A **fixture catalog** in `telco-oran` maps scenario → checked-in TelecomTS window. Each Kafka metrics message is **one sample**. `ran-anomaly-detector` stays the stream client: deserialize → `POST /v1/detect` → publish **typeless** `ran-anomalies` only when AD says anomalous.

**Hybrid serving** matches APPENG-6062: one predictor image (`ran-ml-service`), `TASK=detect`, local `uv run` / fixture weights, **OpenShift AI InferenceService** in cluster. Self-contained artifact (`MLFLOW_MODEL_URI` or path: weights + preprocess + task + labels). Serve-only; training is a later epic.

**Detect is mandatory.** No enable-flag skip on the detector. `/ready` requires the AD predictor. Classify (6062) stays optional so this ticket can demo **click → Mantis AD → LLM** without the 10-class head.

Identity: synthetic **`incident_id`** (chat/Kafka handle, not a tower) plus real TelecomTS metadata (`zone`, application, …). No invented `band`. The model sees only `kpi_window`.

## User Stories

1. As a **network operator**, I want ML to decide whether a TelecomTS window is anomalous, so that detection is grounded in KPI patterns rather than fixed radio thresholds.
2. As a **network operator**, I want only anomalous windows published, so that the dashboard is not flooded with the ~96% normal traffic.
3. As a **network operator**, I want a stable **incident id** in chat, so that I can refer to “this anomaly” without pretending it is a physical cell.
4. As a **network operator**, I want to see real lab context (zone, application, …), so that I understand the trace I am looking at.
5. As a **network operator**, I want Granite to still explain and recommend a vendor-documented fix after AD fires, so that I get actionable guidance before 10-class ML exists.
6. As a **demo presenter**, I want to click a frontend button and inject a known TelecomTS fixture into the live pipeline, so that the demo is reproducible.
7. As a **demo presenter**, I want buttons for labeled scenarios (known anomalous classes plus at least one normal), so that I can show both “AD fires” and “AD stays quiet.”
8. As a **demo presenter**, I want the pitch to say **public 5G lab traces**, so that we do not claim the customer’s live RAN.
9. As a **platform engineer**, I want the AD model deployed as an InferenceService on OpenShift AI, so that the demo showcases RH model serving.
10. As a **platform engineer**, I want the same predictor image to run via `uv run` with `TASK=detect`, so that local and CI do not need a cluster.
11. As a **platform engineer**, I want one predictor image shared with APPENG-6062, so that detect and classify differ only by artifact and InferenceService.
12. As a **platform engineer**, I want artifacts loaded from `MLFLOW_MODEL_URI` or a local path with the **same layout**, so a future training pipeline can swap URIs without rewriting the detector.
13. As a **data scientist**, I want preprocess (feature order, encodings, scaler, 128→512 interpolation) inside the predictor only, so serving matches Alan’s TelecomTS export.
14. As a **data scientist**, I want checked-in fixtures that are known AD-positive (and one normal), so that presenter clicks are deterministic.
15. As an **SRE**, I want the detector **not Ready** when the AD predictor is missing or unready, so that Kubernetes does not send traffic to a detector with no model.
16. As an **SRE**, I want a single failed `POST /v1/detect` to skip/retry that sample without crashing the pod, so that one timeout does not flap the service.
17. As an **SRE**, I want normals never published to `ran-anomalies`, so that downstream RCA is not invoked on healthy windows.
18. As a **Helm operator**, I want the detect InferenceService and detector env wired in the hub chart, so that cluster deploy is one-step.
19. As a **frontend user**, I want Demo Mode buttons to name TelecomTS scenarios, so that the UI matches what is injected.
20. As a **chatbot user**, I want replies to talk about incident id and zone, not cell/band/rule types, so that the assistant matches the new contract.
21. As a **consumer of `ran-anomalies`**, I want a stable typeless schema (`incident_id`, TelecomTS metadata, `kpi_window`, AD score), so that RCA and chat do not depend on rule names.
22. As an **integration test author**, I want JSON Schema on `ran-anomalies` (and updated enriched output), so that contract breaks fail CI.
23. As a **developer**, I want `DETECT_INFERENCE_URL` on the detector, so that classify and detect URLs cannot be confused.
24. As a **developer**, I want unit tests to mock detect HTTP (or talk to fixture weights), so that CI does not load the 8M model for every detector test.
25. As a **maintainer**, I want the six-KPI random factory and rule engine off the happy path, so that two detectors cannot disagree.
26. As a **compliance reviewer**, I want `ad_label` and `ad_confidence` on every published anomaly, so that AD decisions are auditable.
27. As a **downstream RCA author** (APPENG-6062), I want the message **to be** the 18×128 window, so that classify does not re-window or pad.
28. As an **operator of local dev**, I want catalog + detector + predictor + LLM path to run without OpenShift AI, so that I can iterate off-cluster.
29. As a **catalog maintainer**, I want fixtures stored as JSON samples in the repo, so that demos do not fetch HuggingFace at click time.
30. As a **RAN chatbot BFF maintainer**, I want `POST /api/demo/trigger` to publish one JSON sample to the metrics topic, so that injection stays the existing control plane.

### Data and demo injection

- Demo source of truth is **checked-in TelecomTS fixtures**, not the six-KPI `RanKpiRecordFactory` / `SimulationFactory` and not a live HuggingFace fetch.
- Replace that factory’s demo role with a **fixture catalog** in `telco-oran`: scenario name → sample (18×128 window + TelecomTS labels). Include at least one **normal** and one window per RCA class you want to present (anomalous windows must be known AD-positive for Alan’s AD artifact).
- Keep **on-demand only**: `POST /api/demo/trigger` on `ran-chatbot-service`. No background replay Deployment/Job for v1.
- Frontend Demo Mode buttons call that endpoint with the new scenario ids (drop `low_signal` / `cell_outage`).
- Trigger response metadata uses `incident_id` and scenario, not `cell_id` / `band`.

### Kafka metrics

- Topic may stay `ran-combined-metrics`.
- **One message = one TelecomTS sample** (JSON). Not a one-row six-KPI CSV. Not a 10 Hz tick stream. No rolling `MetricsStore` / `history_size`.
- Detector deserializes the sample, calls detect, and on **anomalous** republishes the window on `ran-anomalies`.

### Detector

- Keep `ran-anomaly-detector` as the Kafka-facing service (consumer, `/health`, `/anomalies` buffer, publish).
- Remove `AnomalyDetector` / rule types from the happy path.
- HTTP client to `DETECT_INFERENCE_URL`, `POST /v1/detect` with `kpi_window`.
- **No `MANTIS_ENABLED`** on the detector. If the predictor is unset, unready, or unreachable, **`/ready` fails** (also still require Kafka consumer when that is enabled).
- Per-message detect failure (timeout, 5xx): **do not publish**, log/retry that sample, **do not** crash the pod.
- Do not publish `ad_label=normal`.

### `ran-anomalies` contract (this ticket owns it)

```json
{
  "incident_id": "1842",
  "zone": "C",
  "application": "Twitch",
  "kpi_window": [],
  "ad_label": "anomalous",
  "ad_confidence": 0.94
}
```

- `kpi_window`: 128 timesteps × 18 TelecomTS channels (flat per-timestep objects; exact channel names follow Alan’s export / catalog).
- `incident_id`: synthetic handle for chat and Kafka. **Not** a model feature. **Not** a physical cell. Do not name it `cell_id`.
- `zone` and other TelecomTS labels: real metadata for humans/LLM; **not** concatenated into the detect tensor.
- No `band`, no `anomaly_type`, no rule sentence field.

### Predictor image (shared with APPENG-6062)

- New hub service `ran-ml-service`: FastAPI + PyTorch + Mantis. This ticket lands **`TASK=detect`**, `POST /v1/detect`, `/health`, `/ready` (model loaded).
- Request: `{ "kpi_window": [ /* 128 × 18 */ ] }`. Response includes label (`anomalous` | `normal`), confidence, optional class index.
- Artifact is **self-contained** (weights + preprocess + task + label names). Helm: `MLFLOW_MODEL_URI`. Local/CI: `MANTIS_MODEL_PATH`, same directory layout. Image is code, not weights.
- Same OCI image for `uv run` and the detect InferenceService. APPENG-6062 adds `TASK=classify` and a second InferenceService; do not build a second codebase.
- Serve-only. Future training = new URI, roll the InferenceService.

### OpenShift AI

- Helm: detect InferenceService, custom container, port 8080, `/v1/detect` (not KServe Predict V2, not TGI/vLLM).
- Detector env: `DETECT_INFERENCE_URL` → that service.
- CPU inference is enough for demo scale.

### Downstream so this ticket stands alone (LLM-only RCA)

- `ran-rca-service` graph stays **RAG Retrieval → Analyze** (no classify node here; ADR-0001 still Pydantic state).
- State and enriched records use 6023 identity (`incident_id`, zone/metadata, AD fields). RAG query uses real TelecomTS context, not `{anomaly_type} {band}`.
- Enriched schema drops required `cell_id` / `band` / `anomaly_type`. LLM `root_cause` / `recommended_fix` remain. `ml_*` classify fields are APPENG-6062.
- Chatbot formatting: incident id + zone + AD score + LLM text. Do not say “Cell 42, Band 29.”

### Modules touched (summary)

| Module | Change |
|--------|--------|
| `telco-oran` | Fixture catalog; rule detector off the demo/happy path |
| `ran-ml-service` | **New** predictor image, detect task |
| `ran-anomaly-detector` | JSON sample in; HTTP detect; typeless publish; ready probes predictor |
| `ran-chatbot-service` | Demo trigger publishes catalog sample; chat identity copy |
| `ran-rca-service` | Consume typeless input; RAG/analyze without rule types |
| `ran-frontend` | Scenario buttons |
| `hub/helm` | Detect InferenceService; detector URL |
| `contracts` | `ran-anomalies` schema; update enriched schema identity |
| `Makefile` | Build/test targets for predictor + detector |

### What makes a good test

- Test **observable behavior** at service boundaries: HTTP responses, detection output dicts, Kafka-shaped JSON, JSON Schema, trigger payload/meta.
- Do **not** test tensor math beyond a smoke classify/detect call with fixture weights.
- Prefer **existing seams**. Mock detect HTTP in detector and RCA tests so CI does not need the 8M model.

### Primary test seam (highest — target one main seam)

**Detector orchestration** — the existing `AnomalyDetectionService` process-message/process-input path (today `test_detection.py` asserts output dicts from a CSV blob):

- Input: one JSON TelecomTS sample (catalog fixture or inline equivalent).
- Mock `POST /v1/detect`.
- Assert: anomalous → one `ran-anomalies`-shaped dict (identity + full `kpi_window` + AD fields), JSON Schema valid; normal → empty list; detect HTTP failure → empty list, no exception out of the handler.
- This is the fire-gate. Keep it as the one primary seam.

### Secondary seams (only where the primary cannot cover)

| Seam | What to assert |
|------|----------------|
| **`ran-ml-service` HTTP** (`TASK=detect`) | Fixture `kpi_window` returns `anomalous`/`normal` + confidence; `/ready` after load |
| **Demo catalog / BFF** | `build_demo_*` (today `test_demo.py`) emits the catalog JSON sample and meta (`incident_id`, scenario); unknown scenario still has a default |
| **Detector `/ready`** | Predictor unready → 503 (extend server probe tests) |
| **`ran-rca-service` graph** | `build_graph().ainvoke` on a 6023-shaped anomaly still LLM-enriches (update `test_graph.py` + enriched schema; no classify) |
| **Chatbot formatter** | Incident id + zone + AD line; no cell/band/rule type |

### Prior art

- Detector output-shape tests (`test_detection.py`)
- Detector Kafka handler / server tests
- BFF demo CSV builder tests (`test_demo.py`) — same seam, new payload
- RCA `test_graph.py` + `jsonschema` on enriched records
- Chatbot prompt/formatter unit tests

### Seams check (for implementer / reviewer)

Intended focus is **one primary seam**: detector input sample → (mocked detect) → `ran-anomalies` dict. Catalog, predictor HTTP, ready probe, RCA graph identity, and chat formatter are secondary. Confirm that matches expectations before implementation.

## Out of Scope

- Training, notebooks, GPU jobs, MLflow UI, promote-to-production CI (URI load only).
- 10-class classify node, classify InferenceService, `ml_*` steering (APPENG-6062).
- KServe Predict V2.
- GPU ServingRuntime.
- Background/looping TelecomTS replay producer.
- Synthesizing 18×128 tensors in code instead of fixtures.
- Inventing `band` or mapping zone onto `cell_id`.
- Edge infrastructure workflow (`agent-service` graph).
- AutoRAG or Granite model serving changes.
- RAN frontend work beyond Demo Mode scenario ids/labels.
- Live HuggingFace download at click or deploy time.

## Further Notes

- Parent Jira: APPENG-5050. Sub-task: APPENG-6023. Sibling: APPENG-6062 (classify is optional; this ticket must stand alone).
- Based on Alan’s Mantis/TelecomTS AD work (Google doc on the Jira). AD and RCA are **two fine-tunes**, one backbone.
- Narrative: fixture click → detect InferenceService → typeless `ran-anomalies` → RAG Retrieval + Analyze (Granite) → enriched topic → chatbot. 6062 inserts classify between detect and RAG.
- `incident_id` is a ticket number for the operator, not a cell tower.
